client/me/purchases/payment-methods/existing-credit-card.tsx

```diff
@@ -0,0 +1,406 @@
+				// Optimistically update the toggle
+				queryClient.setQueryData( [ 'tax-info-is-set', storedDetailsId ], {
+					tax_postal_code: onMutateInputValues.tax_postal_code,
+					tax_country_code: onMutateInputValues.tax_country_code,
+					is_tax_info_set: true,
+				} );
+				closeDialog();
+			},
+			onSuccess: ( onSuccessInputValues: {
+				tax_postal_code: string;
+				tax_country_code: string;
+			} ) => {
+				queryClient.setQueryData( [ 'tax-info-is-set', storedDetailsId ], {
+					tax_postal_code: onSuccessInputValues.tax_postal_code,
+					tax_country_code: onSuccessInputValues.tax_country_code,
+					is_tax_info_set: true,
+				} );
+			},
+		}
+	);
+
+	const [ inputValues, setInputValues ] = useState( {
+		tax_postal_code: data?.tax_postal_code ?? '',
+		tax_country_code: data?.tax_country_code ?? '',
+	} );
+
+	const postalCodeValue = inputValues.tax_postal_code;
+	const countryCodeValue = inputValues.tax_country_code;
+
+	const updateTaxInfo = useCallback( () => {
+		mutation.mutate( inputValues );
+	}, [ mutation, inputValues ] );
+
+	const onChangeCountryCode = ( e: { target: { value: string } } ) => {
+		setInputValues( { ...inputValues, tax_country_code: e.target.value } );
+	};
+
+	const onChangePostalCode = ( e: { target: { value: string } } ) => {
+		setInputValues( { ...inputValues, tax_postal_code: e.target.value } );
+	};
+
+	const renderEditForm = (): JSX.Element => {
+		return (
+			<form>
+				<div className="contact-fields payment-methods__tax-fields">
+					<RenderEditFormFields
+						postalCodeValue={ postalCodeValue }
+						countryCodeValue={ countryCodeValue }
+						onChangePostalCode={ onChangePostalCode }
+						onChangeCountryCode={ onChangeCountryCode }
+					/>
+				</div>
+			</form>
+		);
+	};
+
+	const formRender = renderEditForm();
+
+	const showButton = true;
+
+	/* translators: %s is the last 4 digits of the credit card number */
+	const maskedCardDetails = sprintf( _x( '**** %s', 'Masked credit card number' ), last4 );
+
+	const taxInfoDisplay = joinNonEmptyValues( ', ', postalCodeValue, countryCodeValue );
+
+	return (
+		<Fragment>
+			<div>
+				<CardHolderName>{ cardholderName }</CardHolderName>
+				<CardDetails>{ maskedCardDetails }</CardDetails>
+				<span>{ `${ __( 'Expiry:' ) } ${ formatDate( cardExpiry ) }` }</span>
+				{ taxInfoDisplay ? (
+					<span className="existing-credit-card__tax-info-display">
+						<span className="existing-credit-card__tax-info-postal-country">
+							{ taxInfoDisplay }
+						</span>
+						<PaymentMethodEditButton
+							onClick={ () => setIsDialogVisible( true ) }
+							show={ showButton }
+							buttonTextContent={ translate( 'Edit', { textOnly: true } ) }
+							scary={ false }
+							borderless={ true }
+						/>
+					</span>
+				) : (
+					<span className="existing-credit-card__tax-info-display tax-info-incomplete">
+						<PaymentMethodEditButton
+							onClick={ () => setIsDialogVisible( true ) }
+							show={ showButton }
```

Comment:
Again, this probably can be hardcoded to save a variable.

```suggestion
							show={ true }
```

Reply:
Fixed.